MapboxNavigation/NavigationMapView.swift

```diff
@@ -90,6 +90,8 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
     
     var previousLocation: CLLocation?
     
+    var routeProgress: RouteProgress?
```

Comment:
I have reservations about making the map view responsible for storing the most recent route progress object, especially given that two of the three callers to `updateCourseTracking(location:animated:)` are in code that could easily pass in the route progress.

In the spirit of #993, what if we take the RouteMapViewController code that currently calls `updateCourseTracking(location:animated:)` and move it inside `NavigationMapView.progressDidChange(_:)` and a new method that handles foregrounding? At least that way the fact that this class hangs onto a route progress would be better encapsulated.